design/KR/kenward.Rmd

```diff
@@ -0,0 +1,64 @@
+---
+title: how to implement Kenward Roger degree of freedom?
+author: "Liming Li"
+output: html_document
+editor_options:
+  chunk_output_type: console
+---
+
+# Objective
+
+The objective is to design how we can implement Kenward-Roger degree of freedom.
+
+# Current design of mmrm package
+
+In current mmrm package, we provided Satterthwaite degree of freedom, through `df_1d` and `df_md`.
+
+# Design of Kenward-Roger degree of freedom
+
+## update to current functions
+
+* `df_1d` and `df_md` renamed to `df_1d_sat` and `df_md_sat`, do not export them
+* add a new function `df_1d` and `df_md` to include degree of freedom selection
+* implement `df_1d_kr(2)` and `df_md_kr(2)` (suffix 2 means that it is improved Kenward-Roger degree of freedom)
+* consider renaming the `df_1d` and `df_md` to something like `beta_test_1d` and `beta_test_md`?
```

Comment:
yeah maybe in a second step - in the first step we would need to be reverse-compatible / deprecate previous use

Reply:
I wonder, these function names get a bit clunky. Do we need to export them in the first place and could those be turned into S3 generics, like `degrees_of_freedom()`? I think the tidyverse style guideline discourages abbreviations although in this case given the audience, it probably makes sense to use `df()`.

Reply:
yeah `df()` could be nice. not sure about method dispatch here.